Update from OpenAI main

ConnorHeggie opened the pull request:
Brings in a few updates, including whisper large v2 model.

NOTE: I have not tested this

jumon committed:
suppress generating non-timestamp tokens at the beginning (#532)

jongwook committed:
fix to return only the text token ids

jongwook committed:
invoking __call__ instead of forward()

jumon committed:
fix compression ratio function (#561)

jongwook committed:
add large-v2 model

- The "large-v2" model is trained for more epochs with regularization and shows improved performance compared to the previous large.
- It has the same architecture as the original large model.
- When `load_model("large")` is called, the "large-v2" model will be loaded.
- We will soon update the paper regarding this new model.

paulharter committed:
Explicitly closing model file after reading it (#630)

philkuz commented:
This won't be sufficient. You have to update predict.py and a few other places to make cog actually use the new model

altryne committed:
Update Hebrew language code to he per IANA registry (#401)

* Update Hebrew language code to he per IANA registry

Per [IANA registry](https://www.iana.org/assignments/language-subtag-registry/language-subtag-registry), `iw` was deprecated as the code for Hebrew in 1989 and the preferred code is `he`

The correct subtag: 
```
%%
Type: language
Subtag: he
Description: Hebrew
Added: 2005-10-16
Suppress-Script: Hebr
%%
``` 
And the deprecation
```
%%
Type: language
Subtag: iw
Description: Hebrew
Added: 2005-10-16
Deprecated: 1989-01-01
Preferred-Value: he
Suppress-Script: Hebr
%%
```

* Update hebrew ISO code to he

Per discussion, it's ok to make this change without backwards compatibility

chenxwh commented:
just pushed a new version to replicate, with the code now in v2 branch that install whisper repo in cog.yaml to easily keep updated with the latest changes

jongwook committed:
large-v2 figure and arxiv url update

jongwook committed:
saving the qk matrix in the attention module for convenience

jongwook committed:
Revert "saving the qk matrix in the attention module for convenience"

This reverts commit 68e44bd83ce6c3e352f74b266aa39d8b649af9e3.

jongwook committed:
MultiHeadAttention to return qk as well

jongwook committed:
word-level timestamps in Multilingual_ASR notebook

jongwook committed:
torch.concatenate -> torch.cat for compatibility

adamreis committed:
Fix tiny transcribe() docstring typo (#857)

s/successfully/successively, which I believe was the intent.

HennerM committed:
Support batch-dimension in log_mel_spectogram (#839)

mikkovedru committed:
Update README.md (#804)

jongwook committed:
add github action to run pytest

jongwook committed:
allow test_transcribe to run on CPU when CUDA is not available

jongwook committed:
rename GitHub workflow

Commit:
Use ndimage.median_filter instead of signal.medfilter (#812)

For a 30s long audio file which didn't have any silence, ndimage.median_filter took 7s where signa.medfilter took 30s.

Co-authored-by: Umar Farooqi <umar@paystash.com>
Co-authored-by: Jong Wook Kim <jongwook@nyu.edu>
